api/mysagw/models.py

```diff
@@ -4,6 +4,48 @@
 from simple_history.models import HistoricalRecords
 
 
+class UniqueBooleanField(models.BooleanField):
+    """
+    BooleanField that guarantees for one True record and sets others to `False`.
+
+    By default only one record is allowed to be `True`. This can be further configured
+    by providing a list of field names as `together` param.
+
+    Example:
+        ```
+        >>> class Email(models.Model):
+        ...     user = models.ForeignKey(User)
+        ...     email = models.EmailField()
+        ...     default = UniqueBooleanField(together=["user"])
+        ```
+
+        This will enforce two things:
+         1. only allow for one entry where `default == True` per user
+         2. if only one record exists for a given user, it has `default = True`,
```

Comment:
I think for things like this you should also consider the case where multiple entries exist, but none of them are set to `True`. Will we select one randomly? Will we disallow saving the current one as `False`?

Reply:
That's already handled. I've extended the doc string accordingly.

> Will we disallow saving the current one as False?

Exactly. It will behave the same way, as if it's the only record.

This should ensure, that if there are Email records, one (and only one) is set as default.

Reply:
After thinking about this a bit more - I think you're right. Awesome work!